File: renewalMockKata/src/main/java/RealSubscriberService.java
import java.util.Arrays;
import java.util.List;
import java.util.Random;

public class RealSubscriberService implements SubscriberService {
	
	Random random = new Random();


	@Override
	public List<String> getSubscribersThatWillExpireBetweenNowAndDate(int day, int month, int year) {
		if(random.nextBoolean()) {
			return null;
		} else {
			String[] strings = {"mysterious email" + random.nextInt()};
			return Arrays.asList(strings);
		}
	}
}

File: renewalMockKata/csharp/RenewalMock/RealSubscriberService.cs
using System;
using System.Collections.Generic;

namespace Katas
{
    public class RealSubscriberService : SubscriberService
    {
        Random random = new Random();
        public List<string> GetSubscribersThatWillExpireBetweenNowAndDate(int day, int month, int year)
        {
            if (random.NextDouble() < 0.5)
			{
				return null;
			}
			else
			{
                String[] strings = { "mysterious email" + random.Next()};
				return new List<string>(strings);
			}
        }
    }
}
